src/php/Build/Compile/ProjectCompiler.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Build\Compile;
+
+use Phel\Build\Extractor\NamespaceExtractorInterface;
+
+final class ProjectCompiler
+{
+    private NamespaceExtractorInterface $namespaceExtractor;
+
+    private FileCompiler $fileCompiler;
+
+    public function __construct(
+        NamespaceExtractorInterface $namespaceExtractor,
+        FileCompiler $fileCompiler
+    ) {
+        $this->namespaceExtractor = $namespaceExtractor;
+        $this->fileCompiler = $fileCompiler;
+    }
+
+    /**
+     * @return list<CompiledFile>
+     */
+    public function compileProject(array $srcDirectories, string $dest): array
+    {
+        $namespaceInformation = $this->namespaceExtractor->getNamespacesFromDirectories($srcDirectories);
+
+        $result = [];
+        foreach ($namespaceInformation as $info) {
+            $targetFile = $dest . '/' . $this->getTargetFileFromNamespace($info->getNamespace());
+            $targetDir = dirname($targetFile);
+            if (!file_exists($targetDir)) {
+                mkdir($targetDir, 0777, true);
```

Comment:
I am aware that you cannot write a unit test for this class because there is `IO` involved (creation of a directory). It would be necessary to extract this call to an interface and inject that via the constructor. But I didn't want to refactor to that level. In this PR I just wanted to apply the minimum "extract-class-refactoring" to show the basic pattern about our module's design and why.